### PR TITLE
adding model files for platforms notification events

### LINKED ISSUE #90: Platforms Notifications Request
**Is your feature request related to a problem? Please describe.**
When receiving platform notifications from Adyen, I can't seem to find models/structs as part of this library to unmarshal the request body into.

**Describe the solution you'd like**
Ideally, I would like to be able to write something along the lines of:

```
var notification platform.Notification
if err := json.Unmarshal(reqBody, &notification); err != nil {
	return err
}
// do something with the notification
```

**Describe alternatives you've considered**
I've considered writing the models/structs in-house but I'm already using this library for everything else, so it would be awesome if the library could support this. I'm hoping using the Open-API generator won't be too much work. Happy to collaborate and help out if needed.

**Additional context**
When calling the `/testNotificationConfiguration` route, I receive:

```
{
    "eventDate": "2020-11-24T21:22:25+01:00",
    "eventType": "ACCOUNT_HOLDER_CREATED",
    "executingUserKey": "ws_285135",
    "live": false,
    "pspReference": "8816062493452583",
    "content": {
        "accountCode": "1234567890",
        "accountHolderCode": "TestAccountHolder",
        "accountHolderDetails": {
            "address": {
                "city": "Amsterdam",
                "country": "NL",
                "houseNumberOrName": "6-50",
                "postalCode": "1011 DJ",
                "stateOrProvince": "Nord Holland",
                "street": "Simon Carmiggeltstraat"
            },
            "bankAccountDetails": [
                {
                    "accountType": "checking",
                    "bankAccountName": "Primary account",
                    "bankAccountReference": "Ref#000001",
                    "bankAccountUUID": "10000000-0000-0000-0000-000000000001",
                    "bankBicSwift": "BSWFT",
                    "bankCity": "Amsterdam",
                    "bankName": "Bank Name Co",
                    "countryCode": "NL",
                    "currencyCode": "EUR",
                    "iban": "NL00NONE1234123412",
                    "ownerCity": "Amsterdam",
                    "ownerCountryCode": "NL",
                    "ownerHouseNumberOrName": "32\/B",
                    "ownerPostalCode": "1000AA",
                    "ownerState": "Nord Holland",
                    "ownerStreet": "Primary Street",
                    "primaryAccount": true
                },
                {
                    "accountNumber": "300000000",
                    "accountType": "checking",
                    "bankAccountName": "Secondary account",
                    "bankAccountReference": "Ref#000002",
                    "bankAccountUUID": "20000000-0000-0000-0000-000000000002",
                    "bankBicSwift": "BSWFT",
                    "bankCity": "Amsterdam",
                    "bankCode": "10000000",
                    "bankName": "Other Bank Name",
                    "branchCode": "20000000",
                    "countryCode": "NL",
                    "currencyCode": "GBP",
                    "ownerCity": "Amsterdam",
                    "ownerCountryCode": "NL",
                    "ownerHouseNumberOrName": "12\/B",
                    "ownerPostalCode": "1000AB",
                    "ownerState": "Nord Holland",
                    "ownerStreet": "Secondary Street",
                    "primaryAccount": false
                }
            ],
            "bankAggregatorDataReference": "9876543212",
            "businessDetails": {
                "doingBusinessAs": "Submerchant Business Name",
                "legalBusinessName": "Submerchant Co",
                "registrationNumber": "R123456",
                "shareholders": [
                    {
                        "address": {
                            "city": "Amsterdam",
                            "country": "NL",
                            "houseNumberOrName": "11\/A",
                            "postalCode": "9999 AA",
                            "stateOrProvince": "Nord Holland",
                            "street": "Fake Street"
                        },
                        "email": "shareholder.one@adyen.com",
                        "name": {
                            "firstName": "TestFirstNameOne",
                            "gender": "FEMALE",
                            "infix": "van",
                            "lastName": "TestLastNameOne"
                        },
                        "personalData": {
                            "dateOfBirth": "1981-02-27",
                            "documentData": [
                                {
                                    "expirationDate": "2078-02-17",
                                    "issuerCountry": "GB",
                                    "issuerState": "Wales",
                                    "number": "PP1234567",
                                    "type": "PASSPORT"
                                },
                                {
                                    "expirationDate": "2081-10-27",
                                    "issuerCountry": "NL",
                                    "issuerState": "Nord Holland",
                                    "number": "ID1234567",
                                    "type": "ID"
                                }
                            ],
                            "nationality": "Dutch"
                        },
                        "phoneNumber": {
                            "phoneCountryCode": "NL",
                            "phoneNumber": "858888138#1111",
                            "phoneType": "Landline"
                        },
                        "shareholderCode": "10000000-0000-0000-0000-000000000001",
                        "shareholderReference": "SHREF01"
                    },
                    {
                        "address": {
                            "city": "London",
                            "country": "GB",
                            "houseNumberOrName": "22\/B",
                            "postalCode": "W1A 0AX",
                            "stateOrProvince": "England",
                            "street": "Fake Street"
                        },
                        "email": "shareholder.two@adyen.com",
                        "name": {
                            "firstName": "TestFirstNameTow",
                            "gender": "MALE",
                            "infix": "van",
                            "lastName": "TestLastNameTwo"
                        },
                        "personalData": {
                            "dateOfBirth": "1982-03-27",
                            "documentData": [
                                {
                                    "expirationDate": "2079-03-17",
                                    "issuerCountry": "GB",
                                    "issuerState": "England",
                                    "number": "DL1234567",
                                    "type": "DRIVINGLICENSE"
                                },
                                {
                                    "expirationDate": "2082-11-27",
                                    "issuerCountry": "NL",
                                    "issuerState": "Nord Holland",
                                    "number": "VN1234567",
                                    "type": "VISA"
                                }
                            ],
                            "nationality": "British"
                        },
                        "phoneNumber": {
                            "phoneCountryCode": "NL",
                            "phoneNumber": "858888138#2222",
                            "phoneType": "Mobile"
                        },
                        "shareholderCode": "20000000-0000-0000-0000-000000000002",
                        "shareholderReference": "SHREF02"
                    }
                ],
                "taxId": "T123456"
            },
            "email": "testemail@adyen.com",
            "merchantCategoryCode": "9876",
            "metadata": {
                "MetaDataKey2": "MetaDataValue2",
                "MetaDataKey1": "MetaDataValue1"
            },
            "payoutMethods": [
                {
                    "merchantAccount": "MA00001",
                    "payoutMethodCode": "PITC000001",
                    "payoutMethodType": "CardToken",
                    "recurringDetailReference": "RDR000001",
                    "shopperReference": "SR000001"
                }
            ],
            "phoneNumber": {
                "phoneCountryCode": "NL",
                "phoneNumber": "858888138",
                "phoneType": "Landline"
            },
            "webAddress": "https:\/\/adyen.com"
        },
        "accountHolderStatus": {
            "status": "Active",
            "statusReason": "Test Reason",
            "processingState": {
                "disabled": false,
                "processedFrom": {
                    "currency": "EUR",
                    "value": 10
                },
                "processedTo": {
                    "currency": "EUR",
                    "value": 100
                },
                "tierNumber": 2
            },
            "payoutState": {
                "allowPayout": true,
                "payoutLimit": {
                    "currency": "EUR",
                    "value": 100
                },
                "disabled": false,
                "tierNumber": 2
            }
        },
        "description": "Test reference - Test description",
        "legalEntity": "Business",
        "primaryCurrency": "EUR",
        "verification": {
            "accountHolder": {
                "checks": [
                    {
                        "type": "COMPANY_VERIFICATION",
                        "status": "PENDING"
                    }
                ]
            },
            "shareholders": [
                {
                    "checks": [
                        {
                            "type": "IDENTITY_VERIFICATION",
                            "status": "PASSED"
                        },
                        {
                            "type": "PASSPORT_VERIFICATION",
                            "status": "AWAITING_DATA",
                            "requiredFields": [
                                "field name: document, shareholder: 10000000-0000-0000-0000-000000000001, field: AccountHolderDetails.BusinessDetails.Shareholders.Document.document"
                            ]
                        }
                    ],
                    "shareholderCode": "10000000-0000-0000-0000-000000000001"
                },
                {
                    "checks": [
                        {
                            "type": "IDENTITY_VERIFICATION",
                            "status": "FAILED",
                            "summary": {
                                "kycCheckCode": 1,
                                "kycCheckDescription": "Test item1: account holder address is not found"
                            }
                        }
                    ],
                    "shareholderCode": "20000000-0000-0000-0000-000000000002"
                }
            ],
            "bankAccounts": [
                {
                    "checks": [
                        {
                            "type": "BANK_ACCOUNT_VERIFICATION",
                            "status": "AWAITING_DATA",
                            "requiredFields": [
                                "field name: bankBicSwift, field: AccountHolderDetails.BankAccountDetails.bankBicSwift",
                                "field name: bankName, field: AccountHolderDetails.BankAccountDetails.bankName"
                            ]
                        }
                    ],
                    "bankAccountUUID": "10000000-0000-0000-0000-000000000001"
                },
                {
                    "checks": [
                        {
                            "type": "BANK_ACCOUNT_VERIFICATION",
                            "status": "FAILED",
                            "summary": {
                                "kycCheckCode": 1,
                                "kycCheckDescription": "Test item1: account address is not found"
                            }
                        }
                    ],
                    "bankAccountUUID": "20000000-0000-0000-0000-000000000002"
                }
            ],
            "cards": [
                {
                    "checks": [
                        {
                            "type": "CARD_VERIFICATION",
                            "status": "AWAITING_DATA",
                            "summary": {
                                "kycCheckCode": 1,
                                "kycCheckDescription": "Test item1: account address is not found"
                            },
                            "requiredFields": [
                                "field name: cardToken, field: AccountHolderDetails.CardTokens.cardToken"
                            ]
                        }
                    ],
                    "payoutMethodCode": "20000000-0000-0000-0000-000000000001"
                }
            ]
        }
    }
}
```

**Note**
If this already exists in the library, feel free to point me in the right direction. Thanks in advance for all your help!

### PR DESCRIPTION
**Pull request by namanhkapur**
## Summary

- code after discussion here: https://github.com/Adyen/adyen-go-api-library/issues/90
- working in collaboration with @KadoBOT to allow support for platforms notification events
- ran command `node index.js generate -l "go" -s "platforms notifications" -v6` in `adyen-openapi-generator` to generate the model files

## Tested scenarios

**Fixed issue**:  Closes #90

### CODE REVIEW

File: src/platformsnotificationevents/model_account_close_notification.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+ *
+ * The Notification API sends notifications to the endpoints specified in a given subscription. Subscriptions are managed through the Notification Configuration API. The API specifications listed here detail the format of each notification.  For more information, refer to our [documentation](https://docs.adyen.com/platforms/notifications).
+ *
+ * API version: 6
+ * Contact: support@adyen.com
+ */
+
+// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.
+
+package platformsNotifications
```
Comment (deepu105) [inline src/platformsnotificationevents/model_account_close_notification.go:12]: Can we keep folder and package name the same for consistency

**Review by deepu105 — CHANGES_REQUESTED** [review]

**Comment by namanhkapur** [comment]
I'm thinking of adding the README so we know where the models were pulled from. Should I add anything else (i.e. `docs` folder or `api` folder)?

**Comment by deepu105** [comment]
> I'm thinking of adding the README so we know where the models were pulled from. Should I add anything else (i.e. `docs` folder or `api` folder)?

We normally have a Readme inside the src folder https://github.com/Adyen/adyen-go-api-library/blob/develop/src/binlookup/README.md

**Comment by deepu105** [comment]
LGTM. I'll leave it up to @KadoBOT

**Comment by namanhkapur** [comment]
any idea why tests are failing? and any way to rebuild and verify manually?

**Pull request merged**